src/Microsoft.AspNetCore.Hosting/Internal/HostingApplicationData.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.Extensions.FileProviders;
+
+namespace Microsoft.AspNetCore.Hosting.Internal
+{
+    public class HostingApplicationData : IHostingApplicationData
+    {
+        public string ApplicationDataPath { get; set; }
+
+        public IFileProvider ApplicationDataFileProvider { get; set; }
+
+        internal void Initialize(WebHostOptions options)
```

Comment:
Don't pass in WebHostOptions if you're only going to consume one field, pass that value in directly.